llvm/lib/Analysis/ScalarEvolution.cpp

```diff
@@ -6945,7 +6945,16 @@ const ConstantRange &ScalarEvolution::getRangeRef(
         ConstantRange RangeFromOps(BitWidth, /*isFullSet=*/false);
 
         for (const auto &Op : Phi->operands()) {
-          auto OpRange = getRangeRef(getSCEV(Op), SignHint, Depth + 1);
+          ConstantRange OpRange = RangeFromOps;
+          if (auto *S = getExistingSCEV(Op)) {
```

Comment:
Hm, using getExistingSCEV here is going to make results even more query-order dependent than they already are. We can see this in how the `print<scalar-evolution>` results all get better, while the actual transform changes are all worse.

Reply:
Yeah, this was just an attempt to limit regressions for this approach.